src/ast/types.rs

```diff
@@ -22,6 +22,13 @@ pub struct Range {
 }
 
 impl Range {
+    pub fn new(location: Location, end_location: Location) -> Self {
```

Comment:
I agree we should have this. It'd be nice to fix the existing call-sites though... Not sure if you're interested in doing that :D

Reply:
I can bust it out.

Reply:
Got this done. I will fix clippy once you are good with the rest of my code.

Reply:
Thank you! That's no small feat! Really appreciate it!

(Please don't take this as discouragement, solely advice.) In the future, for a change like that, I'd suggest breaking it out into a separate PR, which has a bunch of benefits. For example:

1. Each PR remains clearer in its responsibilities. In the future, when someone digs up this PR to understand the `pyupgrade` changes, they'll need to sift through the `Range` refactor too.
2. Each PR can be merged faster. A PR that just does the `Range` change would be easy to approve and merge, since it'd be decoupled from the `pyupgrade` changes.

Not gonna ask that in this case since it's already done, unless you're eager to try it.